sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -1332,3 +1349,432 @@ class MultiFileCloudOrcPartitionReader(
+      .withNumPyTypes(false)
+      .includeColumn(includedColumns: _*)
+      .build()
+
+    // about to start using the GPU
+    GpuSemaphore.acquireIfNecessary(TaskContext.get())
+
+    val table = withResource(new NvtxWithMetrics("ORC decode", NvtxColor.DARK_GREEN,
+      metrics(GPU_DECODE_TIME))) { _ =>
+      Table.readORC(parseOpts, dataBuffer, 0, dataSize)
+    }
+
+    metrics(NUM_OUTPUT_BATCHES) += 1
+    table
+  }
+
+  /**
+   * Write a header for a specific file format. If there is no header for the file format,
+   * just ignore it and return 0
+   *
+   * @param buffer where the header will be written
+   * @return how many bytes written
+   */
+  override def writeFileHeader(buffer: HostMemoryBuffer): Long = {
+    withResource(new HostMemoryOutputStream(buffer)) { out =>
+      withResource(new DataOutputStream(out)) { dataOut =>
+        dataOut.writeBytes(OrcFile.MAGIC)
+        dataOut.flush()
+      }
+      out.getPos
+    }
+  }
+
+  /**
+   * Writer a footer for a specific file format. If there is no footer for the file format,
+   * just return (hmb, offset)
+   *
+   * Please be note, some file formats may re-allocate the HostMemoryBuffer because of the
+   * estimated initialized buffer size may be a little smaller than the actual size. So in
+   * this case, the hmb should be closed in the implementation.
+   *
+   * @param buffer         The buffer holding (header + data blocks)
+   * @param bufferSize     The total buffer size which equals to size of (header + blocks + footer)
+   * @param footerOffset   Where begin to write the footer
+   * @param blocks         The data block meta info
+   * @param clippedSchema  The clipped schema info
+   * @return the buffer and the buffer size
+   */
+  override def writeFileFooter(
+      buffer: HostMemoryBuffer,
+      bufferSize: Long,
+      footerOffset: Long,
+      stripes: Seq[DataBlockBase],
+      clippedSchema: SchemaBase): (HostMemoryBuffer, Long) = {
+    val lenLeft = bufferSize - footerOffset
+    closeOnExcept(buffer) { _ =>
+      withResource(buffer.slice(footerOffset, lenLeft)) { finalizehmb =>
+        withResource(new HostMemoryOutputStream(finalizehmb)) { rawOut =>
+          // We use the first stripe's ctx
+          // What if the codec is different for the files which need to be combined?
+          val ctx = stripes(0).ctx
+          withCodecOutputStream(ctx, rawOut) { (_, protoWriter, codecStream) =>
+            var numRows = 0L
+            val fileFooterBuilder = OrcProto.Footer.newBuilder
+            // get all the StripeInformation and the total number rows
+            stripes.foreach { stripeWithMeta =>
+              numRows += stripeWithMeta.stripe.infoBuilder.getNumberOfRows
+              fileFooterBuilder.addStripes(stripeWithMeta.stripe.infoBuilder.build())
+            }
+
+            // write the footer
+            val footer = fileFooterBuilder.setHeaderLength(OrcFile.MAGIC.length)
```

Comment:
can we commonize the write footer code with writeOrcOutputFile , seems very close.